packet/icmp.go

```diff
@@ -15,3 +20,27 @@ type ICMP struct {
 	Checksum uint16
 	Data     []byte
 }
+
+func sendICMPPacket(conn *net.Conn, icmp ICMPPacket) (ICMP, TTL, error) {
+	sentBytes, err := (*conn).Write(icmp.Parse())
+	if err != nil {
+		return ICMP{}, 0, err
+	}
+
+	buffer := make([]byte, IP_HEADER_SIZE+sentBytes)
+
+	_, err = (*conn).Read(buffer)
+	if err != nil {
+		return ICMP{}, 0, err
+	}
+
+	ipHeader := buffer[:IP_HEADER_SIZE]
+	icmpHeader := buffer[IP_HEADER_SIZE:]
+
+	return ICMP{
+		Type:     icmpHeader[0],
+		Code:     icmpHeader[1],
+		Checksum: utils.ConcatBytes(icmpHeader[2], icmpHeader[3]),
+		Data:     icmpHeader[4:],
+	}, TTL(ipHeader[8]), nil
```

Comment:
It will be better to export an IP struct as well, instead of single field relevant for now.